sklearn/svm/_base.py

```diff
@@ -219,6 +219,18 @@ def fit(self, X, y, sample_weight=None):
         )
         solver_type = LIBSVM_IMPL.index(self._impl)
 
+        if self._impl in ["c_svc", "nu_svc"]:
+            est_dep = "SVC" if self._impl == "c_scv" else "NuSVC"
+            if self.probability != "deprecated":
+                warnings.warn(
+                    f"parameter `probability` will be deprecated in version 1.8, "
+                    f"use `CalibratedClassifierCV({est_dep}(), ensemble=False)` "
+                    f"instead of `{est_dep}(probability=True)`",
+                    FutureWarning,
+                )
+            else:
+                self.probability = False
```

Comment:
As per scikit-learn estimator API, `fit` should not change the value of a public attribute set in the constructor. Instead, define a local variable named `probability` and set it to `self.probability` when `self.probability != "deprecated"` and `False` otherwise.

Reply:
Then update the rest of the `fit` method to use the `probability` local variable instead of `self.probability`.